src/Plugin/Field/FieldFormatter/FileDownloadLinkFormatter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * @file
+ *  Contains Drupal\file_entity\Plugin\Field\FieldFormatter\FileDownloadLinkFormatter
+ */
+
+namespace Drupal\file_entity\Plugin\Field\FieldFormatter;
+
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Template\Attribute;
+use Drupal\file\Plugin\Field\FieldFormatter\FileFormatterBase;
+
+/**
+ * Plugin implementation of the 'file_download_link' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "file_download_link",
+ *   label = @Translation("Download link"),
+ *   field_types = {
+ *     "file",
+ *     "image"
+ *   }
+ * )
+ */
+class FileDownloadLinkFormatter extends FileFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element['text'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Link text'),
+      '#description' => t('This field supports tokens.'),
+      '#default_value' => $this->getSetting('text'),
+    );
+    // If we have the token module available, add the token tree link.
+    if (\Drupal::moduleHandler()->moduleExists('token')) {
+      $token_types = array('file');
+      if (!empty($form['#entity_type'])) {
+        $token_types[] = $form['#entity_type'];
+      }
+      $element['token_tree_link'] = array(
+        '#theme' => 'token_tree_link',
```

Comment:
This should be #type => token_tree. See token_help() for an example. It now also accepts #dialog => TRUE, which will render it in a dialog.